gui/src/app/components/FileEditor/TextEditor.tsx

```diff
@@ -18,9 +7,27 @@ import {
   useState,
 } from "react";
 
-import monacoAddStanLang from "./monacoStanLanguage";
+import Loading from "@SpComponents/Loading";
+import { UserSettingsContext } from "@SpCore/Settings/UserSettings";
+import { CodeMarker } from "@SpCore/Stanc/Linting";
+import { unreachable } from "@SpUtil/unreachable";
+
+import { Editor, loader, useMonaco, type Monaco } from "@monaco-editor/react";
+import type { editor, IDisposable } from "monaco-editor";
+
 import { ToolBar, ToolbarItem } from "./ToolBar";
 
+import monacoAddStanLang from "./monacoStanLanguage";
+// loader from @monaco-editor/react handles the loading of the monaco editor
+// importantly, it downloads from a CDN, so we need to make sure we
+// only depend on types from the monaco-editor package to avoid
+// downloading twice.
```

Comment:
```suggestion
// importantly, it downloads from a CDN, so we need to make sure our
// dependency on the monaco-editor package is limited to types only,
// to avoid downloading twice.
```